lib/abrt.c

```diff
@@ -444,6 +497,35 @@ occurrences_from_dir(const char *directory,
     return result;
 }
 
+static char *
+get_component(const char *directory,
+              char **error_message)
+{
+    if (is_packaged(directory))
+        return file_contents(directory, "component", error_message);
+
+    /* create component from executable basename concatenated with '-unpackaged' str */
+    if (file_exist(directory, "executable"))
+    {
+        char *executable = file_contents(directory, "executable", error_message);
+        if (executable)
+        {
+            const char *basename = strrchr(executable, '/');
+            if (basename)
+                ++basename;
+            else
+                basename = executable;
+
+            char *artificial_component = sr_asprintf("%s-unpackaged", basename);
```

Comment:
just an aesthetic detail, but maybe would be prettier to have artificial_component as:
somebasename (unpackaged)
or maybe
(unpackaged) somebasename
instead of
somebasename-unpackaged
I think it would be better readable especially in gnome-abrt, since it would be clear on first view, which part is basename and which represents "unpackaged" information